piecedirectory/piecedirectory.go

```diff
@@ -347,61 +349,49 @@ type cachedSectionReader struct {
 	types.SectionReader
 	ps       *PieceDirectory
 	pieceCid cid.Cid
-	// The number of references to the piece reader - only accessed under prCacheLk
-	refs int
 	// Signals when the underlying piece reader is ready
 	ready chan struct{}
 	// err is non-nil if there's an error getting the underlying piece reader
 	err error
 }
 
 func (r *cachedSectionReader) Close() error {
-	r.ps.prCacheLk.Lock()
-	defer r.ps.prCacheLk.Unlock()
-
-	r.refs--
-	if r.refs == 0 {
-		r.ps.pieceReaderCache.Remove(r.pieceCid)
-		return r.SectionReader.Close()
```

Comment:
I think we want to close the underlying section reader here.

Reply:
Closing it now in the expire callback.